File: exercise 2/ColorImage.cpp
#include "opencv2/core/core.hpp"
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include "iostream"

 // Author Rishab Shah

using namespace cv;
using namespace std;
 
int main(int argc, char** argv )
{
    Mat src;
    src = imread(argv[1], CV_LOAD_IMAGE_COLOR);
    namedWindow( "Original image", CV_WINDOW_AUTOSIZE );
    imshow( "Original image", src);

	vector<Mat> input_planes0(3);
	split(src,input_planes0);
	Mat channel1_display, channel2_display, channel3_display;
        imshow("Red",   input_planes0[2]);
        cout<<int(input_planes0[2].at<uchar>(20,25))<<endl;
        imshow("Green",   input_planes0[1]);
        cout<<int(input_planes0[1].at<uchar>(20,25))<<endl;
        imshow("Blue",   input_planes0[0]);
        cout<<int(input_planes0[0].at<uchar>(20,25))<<endl;


	Mat ycrcb_image;
	vector<Mat> input_planes(3);
	cvtColor(src, ycrcb_image, CV_BGR2YCrCb);
	split(ycrcb_image,input_planes);
        imshow("Y",   input_planes[0]);
        cout<<int(input_planes[2].at<uchar>(20,25))<<endl;
        imshow("Cb",   input_planes[1]);
        cout<<int(input_planes[2].at<uchar>(20,25))<<endl;
        imshow("Cr",   input_planes[2]);
        cout<<int(input_planes[2].at<uchar>(20,25))<<endl;


	Mat hsv_image;
	cvtColor(src, hsv_image, CV_BGR2HSV);
	vector<Mat> hsv_planes(3);
	split(hsv_image,hsv_planes);
        imshow("Hue",   hsv_planes[0]);
        cout<<int(hsv_planes[0].at<uchar>(20,25))<<endl;
        imshow("Saturation",   hsv_planes[1]);
        cout<<int(hsv_planes[1].at<uchar>(20,25))<<endl;
        imshow("Value",   hsv_planes[2]);
        cout<<int(hsv_planes[2].at<uchar>(20,25))<<endl;


	
    waitKey(0);                                       
    return 0;
} 
// 235
// 138
// 106
// 96
// 96
// 96
// 7
// 140
// 235

File: exercise 2/ColorImage.py
import cv2
img = cv2.imread("baboon.jpg")
blue,green,red=cv2.split(img)
hsv= cv2.cvtColor(img,cv2.COLOR_BGR2HSV)
Ycrbr=cv2.cvtColor(img,cv2.COLOR_BGR2YCrCb)
y,cr,br=cv2.split(Ycrbr)
hue,saturation,value=cv2.split(hsv)
#cv2.imshow ('Blue.jpg',hsv[:,:,0])
#cv2.imshow ('Green',hsv[:,:,1])
#cv2.imshow ('Red', hsv[:,:,2])

cv2.imwrite("Hue.jpg", hue)
cv2.imwrite ("Saturation.jpg",saturation)
cv2.imwrite("Value.jpg", value)

cv2.imwrite("Y.jpg", y)
cv2.imwrite ("Cr.jpg",cr)
cv2.imwrite("Br.jpg", br)

cv2.imwrite("blue.jpg", blue)
cv2.imwrite ("green.jpg",green)
cv2.imwrite("red.jpg", red)
cv2.waitKey(0) 
cv2.destroyAllWindows()

#brightYCB = cv2.cvtColor(bright, cv2.COLOR_BGR2YCrCb)